src/sstruct_mv/sstruct_matmult.c

```diff
@@ -1097,8 +1099,18 @@ hypre_SStructMatmultComputeU( hypre_SStructMatmultData *mmdata,
          hypre_MatvecCommPkgCreate(parcsr_sP);
       }
 
-      //hypre_BoomerAMGBuildCoarseOperator(parcsr_sP, parcsr_sA, parcsr_sP, &parcsr_uM);
-      parcsr_uM = hypre_ParCSRMatrixRAP(parcsr_sP, parcsr_sA, parcsr_sP);
+      /* Compute unstructured RAP */
+      if (!urap2)
+      {
+         /* Note: this call does NOT add structural zeros to the diagonal */
+         parcsr_uM = hypre_ParCSRMatrixRAPKT(parcsr_sP, parcsr_sA, parcsr_sP, 1, 0);
+      }
+      else
+      {
+         hypre_ParCSRMatrix *Q = hypre_ParCSRMatMat(parcsr_sA, parcsr_sP);
+         parcsr_uM = hypre_ParCSRTMatMatKT(parcsr_sP, Q, 1);
+         hypre_MatvecCommPkgCreate(parcsr_uM);
```

Comment:
I'm curious what the reason for the two different options here is (toggled by `urap2`). Also, why do you create the matvec comm pkg here but not above?

Reply:
I was experimenting with both ways of computing the unstructured RAP on GPUs and how that affect performance. It turns out they are pretty similar, but the current default is slightly faster. I left the `urap2` check there in case in want to revisit this or expose this to users (as we do for BoomerAMG). 

Regarding the commpkg, it turns out that the RAPKT function calls it under the hood, while the MatMatKT doesn't. We should probably unify the behavior here, but I didn't do it in this PR